LottoDiary/LottoDiary/Sources/Present/ChartLottoListView/LottoList/LottoListDataSourceController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  LottoListDataSourceController.swift
+//  LottoDiary
+//
+//  Created by 송선화 on 2022/11/27.
+//
+
+import UIKit
+
+class LottoListDataSourceController {
+    enum Section {
+        case goal, buy, win
+        
+        var title: String {
+            switch self {
+            case .goal:
+                return "목표 금액"
+            case .buy:
+                return "구매 금액"
+            case .win:
+                return "당첨 금액"
+            }
+        }
+        
+        var image: UIImage {
+            switch self {
+            case .goal:
+                return UIImage(named: "medal")!
+            case .buy:
+                return UIImage(named: "happy")!
+            case .win:
+                return UIImage(named: "trophy")!
+            }
+        }
+    }
+    
+    enum GoalResult {
+        case success
+        case fail
+        case percent
+        
+        var title: String {
+            switch self {
+            case .success:
+                return "달성 완료!"
+            case .fail:
+                return "달성 실패!"
+            case .percent:
+                return ""
+            }
+        }
+        
+        var textColor: UIColor {
+            switch self {
+            case .success:
+                return UIColor.designSystem(.mainGreen)!
+            case .fail:
+                return UIColor.designSystem(.mainOrange)!
+            case .percent:
+                return UIColor.clear
+            }
+        }
+    }
+    
+    struct Amount: Hashable {
+        let id = UUID()
+        var amount: Double?
+        var result: GoalResult?
+        var percent: Int?
+    }
```

Comment:
Amount 구조체가 Hashable한 상태이고, id도 UUID로 고유한 식별자로 사용이 되니까 둘 중 하나만 사용해도 될 거 같아.
나같은 경우는 아래처럼 사용하고 있어
```suggestion
    struct Amount: Identifiable {
        let id = UUID().uuidString()
        var amount: Double?
        var result: GoalResult?
        var percent: Int?
    }
```
아래 문서 참고해봐도 좋을거같아.
https://developer.apple.com/documentation/swift/identifiable

Reply:
헐 나는 거기까지 생각하지 못했는데 찾아보니까 id를 identifier로 쓰는게 더 효율적이라고 하더라! 
https://velog.io/@dev_jane/CollectionView-Diffable-DataSource%EC%99%80-Hashable%EC%97%90-%EB%8C%80%ED%95%B4%EC%84%9C
https://velog.io/@dev_jane/UICollectionViewDiffableDataSource%EC%9D%98-identifier%EA%B0%80-Hashable-%ED%95%B4%EC%95%BC%ED%95%98%EB%8A%94-%EC%9D%B4%EC%9C%A0%EA%B0%80-%EB%AD%98%EA%B9%8C

내가 참고했던 글인데 한번 읽어보면 좋을거같아!

Reply:
0cba2bb41a91c1302de0825d19b9d868fcdff056